tests/integration/cattletest/core/test_svc_discovery.py

```diff
@@ -789,6 +789,82 @@ def test_remove_environment_w_active_svcs(super_client,
+    instances = super_client. \
+        list_container(name="compose_" + service.name
+                            + "_" + number, state="running")
+    assert len(instances) == 1
+    return instances[0]
+
+
+def _validate_instance_removed(service, super_client, number):
+    instances = super_client. \
+        list_container(name="compose_" + service.name + "_" + number)
+    assert len(instances) == 1
+    instance = instances[0]
+    wait_for_condition(
+        super_client, instance, _resource_is_removed,
+        lambda x: 'State is: ' + x.state)
+
+
+def _resource_is_removed(resource):
+    return resource.state == 'removed'
+
+
+def test_valiate_service_scaleup_scaledown(super_client,
+                                           admin_client, sim_context, nsp):
+    env = admin_client.create_environment(name="compose")
+    env = admin_client.wait_success(env)
+    assert env.state == "active"
+
+    image_uuid = sim_context['imageUuid']
+    launch_config = {"imageUuid": image_uuid}
+
+    service = super_client.create_service(name=random_str(),
+                                          environmentId=env.id,
+                                          networkId=nsp.networkId,
+                                          launchConfig=launch_config,
+                                          scale=3)
+    service = super_client.wait_success(service)
+    assert service.state == "inactive"
+
+    # activate services
+    env.activateservices()
+    service = admin_client.wait_success(service, 120)
+    assert service.state == "active"
+
+    _validate_compose_instance_start(service, super_client, "1")
+    instance2 = _validate_compose_instance_start(service, super_client, "2")
+    _validate_compose_instance_start(service, super_client, "3")
+
+    # destroy the instance2
+    instance2 = wait_success(super_client, instance2)
+    instance2 = wait_success(super_client, instance2.stop())
+    assert instance2.state == 'stopped'
+
+    instance2 = wait_success(super_client, instance2.remove())
+    assert instance2.state == 'removed'
+
+    # scale up the service
+    service = admin_client.update(service, scale=4)
+    service = admin_client.wait_success(service, 120)
+    assert service.state == "active"
+
+    _validate_compose_instance_start(service, super_client, "1")
+    _validate_compose_instance_start(service, super_client, "2")
+    _validate_compose_instance_start(service, super_client, "3")
+    _validate_compose_instance_start(service, super_client, "4")
+
+    # scale down the service
+    service = admin_client.update(service, scale=2)
+    service = admin_client.wait_success(service, 120)
+    assert service.state == "active"
+    _validate_compose_instance_start(service, super_client, "1")
+    _validate_compose_instance_start(service, super_client, "2")
+    _validate_instance_removed(service, super_client, "3")
+    _validate_instance_removed(service, super_client, "4")
+
+
```

Comment:
Want a test to verify the no-op behavior of updating a deactivated service?
I don't think you _need_ it and its absence wont stop me from LGTMing, but just wanted to point it out.